### PR TITLE
CC-875 update go to 1.21

### PR DESCRIPTION
**Pull request by libmartinito**

### CODE REVIEW

File: starter_templates/go/your_server.sh [hunk 1]
```diff
@@ -6,6 +6,5 @@
 #
 # DON'T EDIT THIS!
 set -e
-tmpFile=$(mktemp)
-go build -o "$tmpFile" app/*.go
-exec "$tmpFile" "$@"
+go build -o "/tmp/codecrafters-http-target/your_server" app/*.go
+exec "/tmp/codecrafters-http-target/your_server" "$@"
```
Comment (libmartinito) [inline starter_templates/go/your_server.sh:10]: used 'codecrafters-http-target' as it's the same format we use in our java starter file. is it fine to keep this format everywhere?

File: dockerfiles/go-1.21.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+ENV CODECRAFTERS_DEPENDENCY_FILE_PATHS="go.mod,go.sum"
+
+WORKDIR /app
+
+COPY go.mod go.sum ./
+
+# Starting from Go 1.20, the go standard library is no loger compiled
+# setting the GODEBUG environment to "installgoroot=all" restores the old behavior
+RUN GODEBUG="installgoroot=all" go install std
```
Comment (libmartinito) [inline dockerfiles/go-1.21.Dockerfile:11]: does something like this work?

Comment (rohitpaulk) [reply dockerfiles/go-1.21.Dockerfile:11]: @libmartinito yep!

Comment (rohitpaulk) [reply dockerfiles/go-1.21.Dockerfile:11]: Nice find

**Pull request merged**